flows-advanced/parent-orchestrator/orchestrating-worker-flows.py

```diff
@@ -0,0 +1,221 @@
+    """Async task to create a flow run from a deployment
+    Args:
+        deployment_id (str): Prefect Deployment ID of a flow deployment
+        params (Dict): the parameters an instance of the subflow needs to run
+        tags (Iterable[str]): tags to apply to the worker flow runs
+    """
+    logger = get_run_logger()
+    try:
+        with await get_client() as client:
+            flow_run_model = await client.create_flow_run_from_deployment(
+                parameters=params, deployment_id=deployment_id, tags=tags
+            )
+            logger.info(f"Created flow run {flow_run_model.flow_run.name}!")
+    except (PrefectHTTPStatusError, ObjectNotFound) as err:
+        logger.error(f"{err!r}")
+        raise
+
+
+@task(name="Poll for subflow status")
+async def poll_for_subflow_completion(
+    filter: FlowRunFilter, POLL_INTERVAL_S: int = 20
+) -> Dict:
+    """Task for monitoring state of worker flow runs and raising their failures
+    Args:
+        filter (FlowRunFilter): Prefect Orion database filter for
+            retrieving worker flow runs.
+        POLL_INTERVAL_S (int, optional): how often to poll for `subflow_status`.
+    Raises:
+        ValueError: if no worker flow runs match `FlowRunFilter` criteria
+    Returns:
+        Dict: summary of worker flow run states
+    """
+    logger = get_run_logger()
+
+    async with get_client() as client:
+        while True:
+            subflow_status = {}
+            subflows = await client.read_flow_runs(flow_run_filter=filter)
+
+            for subflow in subflows:
+                subflow_status.setdefault(subflow.state.name, []).append(subflow)
+
+            if not subflow_status:
+                raise ValueError("No subflows matching flow run filter!")
+
+            if "Failed" in subflow_status:
+                message = (
+                    "The following worker flow runs finished in state Failed: "
+                    f"{[run.name for run in subflow_status['Failed']]}"
+                )
+                logger.warning(message)
+
+            if "Crashed" in subflow_status:
+                message = (
+                    "The following worker flow runs finished in state Crashed: "
+                    f"{[run.name for run in subflow_status['Crashed']]}"
+                )
+                logger.warning(message)
+
+            logger.info(summarize(subflow_status))
+
+            if not any(
+                state in subflow_status for state in ["Pending", "Scheduled", "Running"]
+            ):
+                return subflow_status
+
+            await asyncio.sleep(POLL_INTERVAL_S)
+
+
+@flow(name="My Orchestrator Flow")
+def orchestrator(worker_deployment_id_block_name: str, chunk_size: int = 2):
+    """Orchestrator flow to kick off instances of worker subflows
+
+    You could also pass in:
+    - a SQL query / filepath / http request info to fetch data to pass to worker flows
```

Comment:
Probably don't want to suggest a sql query as a parameter as it opens the door for some injection

Reply:
okay, what about the name of a block that points at a `String` that holds a SQL query